### ISSUE TITLE
Exclude option not working if no staged_files use

### ISSUE THREAD

**Issue by ctjhoa** [evilmartians/lefthook#441]
### :wrench: Summary

I want to exclude files from a lefthook command but I do not use files variable in my `run` command line. 

### Lefthook version

`1.2.7 c88a009cfa095e97ce47f8c5d5e1db5e6c844341`

### Steps to reproduce

```yml
# lefthook.yml
pre-commit:
  parallel: true
  commands:
    lint:
      tags: lint projects
      exclude: 'apps/memory-360.*'
      glob: '*.{ts,tsx,js,jsx,rb,ru}'
      run: yarn affected:lint --uncommitted
```

### Expected results

`run` should not be executed when an excluded file is affected.

### Actual results

`run` is always executed.

### Possible Solution

I found a workaround by turning my lefthook config into this
```yml
# lefthook.yml
pre-commit:
  parallel: true
  commands:
    lint:
      tags: lint projects
      exclude: 'apps/memory-360.*'
      glob: '*.{ts,tsx,js,jsx,rb,ru}'
      run: yarn affected:lint --uncommitted && echo {staged_files} > /dev/null
```

### Screenshots (if appropriate)

Actual results
![2023-03-03-170524_1278x113_scrot](https://user-images.githubusercontent.com/1716173/222768911-7ea2c31d-32d1-4fe9-b043-417af91b20d6.png)

With workaround

![2023-03-03-170501_1269x335_scrot](https://user-images.githubusercontent.com/1716173/222768907-23e29b7f-425d-4f4e-8e2a-3c99564e5c0d.png)

**Comment by mrexox** [comment]
Hi! This issue totally makes sense, but things are a bit complicated.

You have `pre-commit` hook  and a command that doesn't reference any files template and doesn't provide a [`files`](https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#files) option. It is fine but when you have `glob` and `exclude` lefthook should decide: what files should the filters be applied to? for `pre-commit` it is `{staged_files}`, For `pre-push` it is `{push_files}`, and what is it for other hooks? Not so easy to answer. We've got a [lot of them](https://github.com/evilmartians/lefthook/blob/master/internal/config/available_hooks.go#L11-L39).

That was the reason this feature wasn't implemented. But I think it makes sense to have it with these requirements:

- for `pre-commit` filter `{staged_files}`
- for `pre-push` filter `{push_files}`
- for all the others - leave the current behavior (no filtering)

Does it sound good?

The only thing I have concerns about - this approach is not verbose because it is implicit and the logic is not obvious. If you have a better suggestion I can consider it and add this feature to the next release. I don't want to bring a breaking change and unexpected behavior. I also don't want to bring non-intuitive features that could be redesigned in the future when people get used to them (like the way `skip` option is designed which I'm afraid to touch/change). How do you think this could be configured the best way? Does using of implicit templates feel intuitive enough?

**Comment by ctjhoa** [comment]
I created this bug ticket for the sake of:
- Confirming the issue
- Avoiding debugging time waste for other people
- Giving workaround

If the existing behavior make sense from a technical point of view / project maintenance and lefthook users have workaround, this limitation can be documented as is.
The real issue here it's that it tooks me 2 hours to find out what's going on :P

**Comment by mrexox** [comment]
I've released this feature in **v1.3.6**. Please, check it out :rocket:
